packages/kbn-docs-utils/src/api_docs/count_eslint_disable.ts

```diff
@@ -6,29 +6,64 @@
  * Side Public License, v 1.
  */
 
-import { execSync } from 'child_process';
+import Fs from 'fs';
+import Path from 'path';
 
 export interface EslintDisableCounts {
   eslintDisableLineCount: number;
   eslintDisableFileCount: number;
 }
 
-export async function countEslintDisableLine(path: string): Promise<EslintDisableCounts> {
-  const disableCountOutputs = await Promise.all([
-    execSync(`grep -rE 'eslint-disable-next-line|eslint-disable-line' ${path} | wc -l`),
-    execSync(`grep -rE 'eslint-disable ' ${path} | wc -l`),
-  ]);
-  const eslintDisableLineCount = Number.parseInt(disableCountOutputs[0].toString(), 10);
-
-  if (eslintDisableLineCount === undefined || isNaN(eslintDisableLineCount)) {
-    throw new Error(`Parsing ${disableCountOutputs[0]} failed to product a valid number`);
+async function fetchAllFilePaths(path: string): Promise<string[]> {
+  if ((await Fs.promises.stat(path)).isFile()) {
+    return [path];
+  }
+  const filePaths: string[] = [];
+  const dirContent = await Fs.promises.readdir(path, { withFileTypes: true });
+  for (const item of dirContent) {
+    const itemPath = Path.resolve(path, item.name);
+    if (item.isDirectory()) {
+      filePaths.push(...(await fetchAllFilePaths(itemPath)));
+    } else if (item.isFile()) {
+      filePaths.push(itemPath);
+    }
   }
+  return filePaths;
+}
 
-  const eslintDisableFileCount = Number.parseInt(disableCountOutputs[1].toString(), 10);
+function findOccurrences(fileContent: string, regexp: RegExp): number {
+  // using the flag 'g' returns an array of found occurrences.
+  const matchingResults = fileContent.toString().match(new RegExp(regexp, 'g')) || [];
+  return matchingResults.length;
+}
 
-  if (eslintDisableFileCount === undefined || isNaN(eslintDisableFileCount)) {
-    throw new Error(`Parsing ${disableCountOutputs[1]} failed to product a valid number`);
-  }
+async function countEsLintDisableInFile(path: string): Promise<EslintDisableCounts> {
+  const fileContent = await Fs.promises.readFile(path, { encoding: 'utf8' });
```

Comment:
I'm not opposed to reading the files, but many of the files have already been read and would be available from the ts project, if we wanted to shave a little bit of time off the script.

Reply:
I took a look at this suggestion, and it feels like a big refactor at this point. I created #140000 to tackle it in a separate issue. I hope it makes sense 😇 

However, I used `asyncMapWithLimit` when calling this method to be easier on memory usage.